升级到beego@v1.12.2，commentRouter出现unknown escape sequence报错

升级从 beego@v1.12.1 升级到beego@v1.12.2，生成的commentsRouter的正则出现错误，编译时报错："unknown escape sequence"。

beego@v1.12.3也同样存在这个问题


```
routers/commentsRouter_controllers_xxxxxxxxx.go:49:28: unknown escape sequence

 46     beego.GlobalControllerRouter["someDir:Controller"] = append(beego.GlobalControllerRouter["someDir:Controller"],
 47         beego.ControllerComments{
 48             Method: "SomeMethod",
 49             Router: "/:id(\d+)",
 50             AllowHTTPMethods: []string{"POST"},
 51             MethodParams: param.Make(),
 52             Filters: nil,
 53             Params: nil})

```


有可能是这个变更引起的
```
diff pkg/mod/github.com/astaxie/beego@v1.12.1/parser.go pkg/mod/github.com/astaxie/beego@v1.12.2/parser.go
503c503
<             ` + "Router: `" + c.Router + "`" + `,
---
>             ` + `Router: "` + c.Router + `"` + `,
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,1 +1,3 @@
 # developing
+- Fix `unknown escape sequence` in generated code. [4385](https://github.com/beego/beego/pull/4385)
+- Using fixed name `commentRouter.go` as generated file name. [4385](https://github.com/beego/beego/pull/4385)

--- a/server/web/parser.go
+++ b/server/web/parser.go
@@ -50,7 +50,6 @@ func init() {
 
 var (
 	lastupdateFilename = "lastupdate.tmp"
-	commentFilename    string
 	pkgLastupdate      map[string]int64
 	genInfoList        map[string][]ControllerComments
 
@@ -71,16 +70,13 @@ var (
 	}
 )
 
-const commentPrefix = "commentsRouter_"
+const commentFilename   = "commentsRouter.go"
 
 func init() {
 	pkgLastupdate = make(map[string]int64)
 }
 
 func parserPkg(pkgRealpath string) error {
-	rep := strings.NewReplacer("\\", "_", "/", "_", ".", "_")
-	commentFilename, _ = filepath.Rel(AppPath, pkgRealpath)
-	commentFilename = commentPrefix + rep.Replace(commentFilename) + ".go"
 	if !compareFile(pkgRealpath) {
 		logs.Info(pkgRealpath + " no changed")
 		return nil
@@ -102,7 +98,10 @@ func parserPkg(pkgRealpath string) error {
 					if specDecl.Recv != nil {
 						exp, ok := specDecl.Recv.List[0].Type.(*ast.StarExpr) // Check that the type is correct first beforing throwing to parser
 						if ok {
-							parserComments(specDecl, fmt.Sprint(exp.X), pkg.PkgPath)
+							err = parserComments(specDecl, fmt.Sprint(exp.X), pkg.PkgPath)
+							if err != nil {
+								return err
+							}
 						}
 					}
 				}
@@ -500,7 +499,8 @@ func genRouterCode(pkgRealpath string) {
     beego.GlobalControllerRouter["` + k + `"] = append(beego.GlobalControllerRouter["` + k + `"],
         beego.ControllerComments{
             Method: "` + strings.TrimSpace(c.Method) + `",
-            ` + `Router: "` + c.Router + `"` + `,
+
+            ` + "Router: `" + c.Router + "`" + `,
             AllowHTTPMethods: ` + allmethod + `,
             MethodParams: ` + methodParams + `,
             Filters: ` + filters + `,

